sriovnet_switchdev.go

```diff
@@ -226,6 +226,44 @@ func findNetdevWithPortNameCriteria(criteria func(string) bool) (string, error)
 	return "", fmt.Errorf("no representor matched criteria")
 }
 
```

Comment:
the added methods are very similar.

i wonder if its worth adding a common method

`getPortFromRepresentor(repNetDev string) (PortFlavour, int, err) `

you can use parseIndexFromPhysPortName , passing it the correct regex according to the port flavor.

note: IMO you can assume this method will not run on old kernels where VF phys port name was just a number

then each of the defined methods below would just call `getPortFromRepresentor` WDYT ?

Reply:
I was thinking the same thing. I don't see context where having the implicit port type check would be useful, so I'll refactor as a single function.